common/spec/dependabot/config/ignore_condition_spec.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+RSpec.describe Dependabot::Config::IgnoreCondition do
+  let(:dependency_name) { "test" }
+  let(:versions) { nil }
+  let(:update_types) { nil }
+  let(:ignore_condition) do
+    described_class.new(
+      dependency_name: dependency_name,
+      versions: versions,
+      update_types: update_types
+    )
+  end
+
+  describe "#versions" do
+    subject(:ignored_versions) { ignore_condition.ignored_versions(dependency) }
+    let(:dependency) do
+      Dependabot::Dependency.new(
+        name: dependency_name,
+        requirements: [],
+        package_manager: "npm_and_yarn",
+        version: "1.2.3"
+      )
+    end
+
+    def expect_allowed(*versions)
+      req = Gem::Requirement.new(ignored_versions.flat_map { |s| s.split(",").map(&:strip) })
```

Comment:
This should work but might be wrong?
```suggestion
      req = Gem::Requirement.new(*ignored_versions) })
```

Reply:
Oh wait it's `Gem::Requirement` so won't work 👍 

We could get the requirement class for the dependency?

Reply:
I figured `Gem::Requirement` would be good enough for a test?

My goal was to write tests as "X should be ignored, Y should not be", as I found that easier to reason about than `"> a.b.c, < d.e.f"`.

Reply:
I threw in a comment aiming to clarify these are limited tools for testing.

TBH they were only really necessary early in development when the ranges were blowing my mind - do you think we should just remove the tests?

Reply:
> My goal was to write tests as "X should be ignored, Y should not be", as I found that easier to reason about than `"> a.b.c, < d.e.f"`.

👍 I agree this makes it easier to understand the tests. I think they are worth keeping!